### ISSUE TITLE
Go to definition is not working on nvim + nvim-lspconfig

### ISSUE THREAD

**Issue by rafaeldelboni** [clojure-lsp/clojure-lsp#901]
**Describe the bug**
Go to definition is not working on nvim + nvim-lspconfig

**To Reproduce**
1. Open a clj file
2. Navigate to a Java FN
3. Execute `:lua vim.lsp.buf.definition()`

**Expected behavior**
Go to definition to java methods should work.

**Screenshots**

<details>
 <summary><b>Log - clojure-lsp</b></summary>
<pre>
2022-04-01T13:32:50.212Z  INFO [clojure-lsp.feature.java-interop:183] - [93m[Java][0m Local JDK source not found.
2022-04-01T13:32:50.212Z  INFO [clojure-lsp.feature.java-interop:188] - [93m[Java][0m Found local JDK source URI, extracting to global LSP cache dir...
2022-04-01T13:32:50.212Z  ERROR [taoensso.timbre:828] - Uncaught exception on thread: async-dispatch-2
[37mcom.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine[m  [32m      PosixJavaThreads.java:  194[m
            [37mcom.oracle.svm.core.thread.JavaThreads.threadStartRoutine[m  [32m           JavaThreads.java:  597[m
                                                 [37mjava.lang.Thread.run[m  [32m                Thread.java:  829[m
                                                                  [37m...[m  [32m                                 [m
   [33mclojure.core.async.impl.concurrent/counted-thread-factory/reify/[1;33mfn[m  [32m             concurrent.clj:   29[m
                   [37mjava.util.concurrent.ThreadPoolExecutor$Worker.run[m  [32m    ThreadPoolExecutor.java:  628[m
                    [37mjava.util.concurrent.ThreadPoolExecutor.runWorker[m  [32m    ThreadPoolExecutor.java: 1128[m
                                                                  [37m...[m  [32m                                 [m
                                   [33mclojure-lsp.handlers/initialize/[1;33mfn[m  [32m               handlers.clj:  115[m
         [33mclojure.core.async.impl.ioc-macros/[1;33mrun-state-machine-wrapped[m  [32m             ioc_macros.clj:  982[m
                 [33mclojure.core.async.impl.ioc-macros/[1;33mrun-state-machine[m  [32m             ioc_macros.clj:  978[m
             [33mclojure-lsp.handlers/initialize/fn/[1;33mstate-machine--auto--[m  [32m               handlers.clj:  115[m
          [33mclojure-lsp.handlers/initialize/fn/state-machine--auto--/[1;33mfn[m  [32m               handlers.clj:  116[m
    [33mclojure-lsp.feature.java-interop/[1;33mretrieve-jdk-source-and-analyze![m  [32m           java_interop.clj:  189[m
                                                    [33mbabashka.fs/[1;33munzip[m  [32m                    fs.cljc:  876[m
                                   [37mjava.nio.file.Files.newInputStream[m  [32m                 Files.java:  156[m
                  [37mjava.nio.file.spi.FileSystemProvider.newInputStream[m  [32m    FileSystemProvider.java:  420[m
                                   [37mjava.nio.file.Files.newByteChannel[m  [32m                 Files.java:  422[m
                                   [37mjava.nio.file.Files.newByteChannel[m  [32m                 Files.java:  371[m
                     [37msun.nio.fs.UnixFileSystemProvider.newByteChannel[m  [32mUnixFileSystemProvider.java:  219[m
[1;31mjava.nio.file.NoSuchFileException[m: [3m/home/delboni/Workspaces/180s/sarue/https:/raw.githubusercontent.com/clojure-lsp/jdk-source/main/openjdk-19/reduced/source.zip[m
</pre>
</details>

**User details (please complete the following information):**
 - OS: ArchLinux 5.17.1-arch1-1 
  - Editor: NVIM v0.6.1
 - Version: (post the result of `clojure-lsp --version`)
```
clojure-lsp 2022.03.31-20.00.20
clj-kondo 2022.03.10-SNAPSHOT
```

**Comment by ericdallo** [comment]
@rafaeldelboni go to definition is not working at all or only for java?

**Comment by rafaeldelboni** [comment]
@ericdallo Just for java, clojure ones are working fine :)

**Comment by ericdallo** [comment]
@rafaeldelboni made some improvements on master, could you try a nightly build?

**Comment by rafaeldelboni** [comment]
I tested the nightly now, and I'm getting this on server startup:
```
2022-04-04T13:23:16.629Z  WARN [clojure-lsp.feature.java-interop:232] - [Java] Skipping download JDK source, setting `:java :download-jdk-source?` is disabled.
2022-04-04T13:23:16.629Z  WARN [clojure-lsp.feature.java-interop:239] - [Java] JDK Source not found, skipping java analysis.
```
When I try a go to definition to a Java Method, I get no errors, but I think is expected because is skipping the java analysis.

**Comment by ericdallo** [comment]
Yes, that means clojure-lsp didn't find any java JDK source in your OS, [this is the waterfall it tries](https://github.com/clojure-lsp/clojure-lsp/blob/master/lib/src/clojure_lsp/feature/java_interop.clj#L87-L97), if that is not found, clojure-lsp will try to download only if `:java :download-jdk-source?` setting is enabled (disabled by default)

**Comment by rafaeldelboni** [comment]
Interesting I've set this in my `~/.lsp/config.edn`
```
{:java {:home "/usr/lib/jvm/java-11-openjdk"}}
```

**Comment by ericdallo** [comment]
There is no such config like `:java :home` (yet ) 😅

**Comment by rafaeldelboni** [comment]
I will add now, I use a [archlinux-java](https://wiki.archlinux.org/title/Java) so I don't like setting this manually, but for this feature I will add here. The config on `.lsp/config.edn` would be a great addition :)

**Comment by rafaeldelboni** [comment]
I've added the `JAVA_HOME`, but still getting:
```
2022-04-04T13:45:09.231Z  WARN [clojure-lsp.feature.java-interop:232] - [Java] Skipping download JDK source, setting `:java :download-jdk-source?` is disabled.
2022-04-04T13:45:09.231Z  WARN [clojure-lsp.feature.java-interop:239] - [Java] JDK Source not found, skipping java analysis.
```

![image](https://user-images.githubusercontent.com/1683898/161557580-ccd0628b-fd48-4be0-b81c-aa360484201a.png)

**Comment by rafaeldelboni** [comment]
Uhm `ls` inside the lib I get this:
![image](https://user-images.githubusercontent.com/1683898/161558137-40069f3e-49b5-478c-beea-ec12f579224f.png)
But no .zip in the folder:
![image](https://user-images.githubusercontent.com/1683898/161558281-8c1792c2-0e32-41dc-a1ef-18f2aef7b7a7.png)

**Comment by rafaeldelboni** [comment]
Ah I think Arch users need to install this package as well to it work:
https://archlinux.org/packages/extra/x86_64/openjdk11-src/

**Comment by ericdallo** [comment]
Yes, probably that is necessary, maybe we should document that somewhere?

**Comment by ericdallo** [comment]
FYI @dharrigan

**Comment by rafaeldelboni** [comment]
Good news is:
```
2022-04-04T13:56:41.941Z  INFO [clojure-lsp.feature.java-interop:215] - [Java] Automatically found local JDK source zip at /usr/lib/jvm/java-11-openjdk/lib/src.zip, extracting to global LSP cache dir...
2022-04-04T13:56:41.956Z  INFO [lsp4clj.core:298] - [Server] Initialized!
2022-04-04T13:56:42.068Z  INFO [clojure-lsp.feature.file-management:28] - :did-open 110ms
2022-04-04T13:56:43.138Z  INFO [clojure-lsp.feature.clojuredocs:23] - [Clojuredocs] Refreshing clojuredocs cache took 1198ms.
2022-04-04T13:56:45.446Z  INFO [clojure-lsp.feature.java-interop:134] - [Java] Analyzing JDK source with clj-kondo took 1776ms
2022-04-04T13:56:45.607Z  INFO [clojure-lsp.feature.java-interop:238] - [Java] JDK Source analyzed successfully.
```
Bad news is that the go to definition still doesn't work, but now looks like a nvim problem, because the logs says this:
```
2022-04-04T13:57:05.331Z  INFO [clojure-lsp.handlers:179] - :definition 359ms
```

**Comment by rafaeldelboni** [comment]
It worked actually hahaha, I was just trying to do a go to definition in a method of a class, in a static one it worked.

**Comment by ericdallo** [comment]
Yep, this is expected as we don't have location support yet from clj-kondo side

**Comment by rafaeldelboni** [comment]
Sure! Thanks for this incredible work and help.

**Comment by ericdallo** [comment]
https://github.com/clojure-lsp/clojure-lsp/issues/907
